### ISSUE TITLE
UPS-RS sequence matching

### ISSUE THREAD

**Issue by janossch** [dcm4che/dcm4chee-arc-light#3567]
I feel this is more like an issue than just a simple question. Could you please check this?

### Discussed in https://github.com/dcm4che/dcm4chee-arc-light/discussions/3493

<div type='discussions-op-text'>

<sup>Originally posted by **janossch** December 14, 2021</sup>
Hello to everyone!

My question is related to [querying workitems](https://github.com/dcm4che/dcm4chee-arc-light/issues/2294) via UPS-RS. 

Let's imagine a workitem and let's suppose there are two items/devices in the Scheduled Station Name Code Sequence.
For sake of simplicity I made it a bit shorter:

```
{
...
 "00404025": {
            "00080100": {
                        "Value": [ "DEVICE1" ]
                    },
            "00080100": {
                        "Value": [ "DEVICE2" ]
                    },
 }
...
}
```

This query returns back the above workitem
_http://localhost:8082/dcm4chee-arc/aets/DCM4CHEE/rs/workitems?00404025.00080100=DEVICE1_

However the result of this query is **204 No Content**
_http://localhost:8082/dcm4chee-arc/aets/DCM4CHEE/rs/workitems?00404025.00080100=DEVICE2_

If I change the order of the items, then DEVICE2 value gives a match and DEVICE1 returns 204. 
It seems [sequence matching](https://dicom.nema.org/medical/dicom/current/output/html/part04.html#sect_C.2.2.2.6) is kind of not working.

Am I missing something here? Could you please help me out?
Thank you!
                    </div>

**Comment by gunterze** [comment]
You are right! According [DICOM PS 3.3, C.30.2 Unified Procedure Step Scheduled Procedure Information Module](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.30.2.html), the Sequences:

Attribute Name | Tag | Attribute Description
-- | -- | --
Scheduled Station Name Code Sequence | (0040,4025) | Identifying  names within the enterprise of the equipment for which the Procedure  Step is scheduled. The names conveyed in Code Value (0008,0100) may be  the same as the AE Titles, but do not have to be.                                                                Zero or more Items shall be included in this Sequence.
Scheduled Station Class Code Sequence | (0040,4026) | Classes of the equipment for which the Procedure Step is scheduled.                                                                Zero or more Items shall be included in this Sequence.
Scheduled Station Geographic Location Code Sequence | (0040,4027) | Geographic locations of the equipment for which the Procedure Step is scheduled.                                                                Zero or more Items shall be included in this Sequence.
 
may include more than one item.

Only Sequence:

Attribute Name | Tag | Attribute Description
-- | -- | --
Scheduled Workitem Code Sequence | (0040,4018) | Coded description of the Procedure Step.                                                                Only a single Item shall be included in this Sequence.

is restricted to include only a single Item.
